jOOQ/src/main/java/org/jooq/tools/Convert.java

```diff
@@ -494,16 +494,31 @@ else if (fromClass == byte[].class) {
                     return convert(Arrays.toString((byte[]) from), toClass);
                 }
                 else if (fromClass.isArray()) {
+                    Object[] fromArray = (Object[]) from;
+
+                    if (Collection.class.isAssignableFrom(toClass) &&
+                            toClass.isAssignableFrom(ArrayList.class)) {
+                        return (U) new ArrayList(Arrays.asList(fromArray));
+                    }
+                    else if (Collection.class.isAssignableFrom(toClass) &&
+                            toClass.isAssignableFrom(LinkedHashSet.class)) {
+                        return (U) new LinkedHashSet(Arrays.asList(fromArray));
+                    }
```

Comment:
I like this implementation approach. Very cunning. Could you perhaps add a comment with an issue reference to the first `if`

```java
// [#3062] Default collections if no specific collection type was requested